GoogleForCreators/web-stories-wp
# Dashboard: Catch Saved Templates up to existing structures

## Summary
Saved templates is not functional and still hidden behind a feature flag but the structure of the page was lagging behind modifications we've made to the other dashboard pages and i wanted to dust this off so that when it's time to implement it is a cleaner process. 

## Relevant Technical Choices

- Mimicking the same directory structure for the Saved Templates view as templates and my stories (Content and Header) 
- Add a serializer to shape saved template data since the API is returning data back to us but it's not populating the story portion yet. This is known and there's a ticket somewhere for this. By serializing the data we can prep the view better. 
- Minor tweak to template state to shape saved templates in the same manor that other serialized data is. 
- Checking that pages has a length before rendering a preview in the dialog once "see details" on the "template" is clicked to prevent an error - figure this is a pretty ok failsafe for bad data that somehow sneaks through. On stories this should never happen because when pages is an array of 0 we don't allow the story to get rendered. 
- Mimic story content structure for context menu and handleMenuItemSelected. Should make adds easier when it's time to build out this view. 
- Gave saved templates it's own "grid", i'm going to follow this up with a move of view grids into the actual view folder. At this point, they are all different enough that "sharing" it isn't really helping. Once we have the structures tightened up and we know what's the same we can share components. 

## To-do

- this is purely a refactor of existing structures! We still have work to do here - I made it a point to not add any functionality to this page that didn't exist before the refactor. The only exception here is that I did make the preview functionality work, it was a pretty straightforward copy and paste from either templates or stories. If text or content or sort or filtering or searching doesn't work that is accurate. 
- I also did not write karma tests since we don't actually know how the APIs here will work (there's a question out about bookmarking and what is "saved") 

## User-facing changes
- None. Saved Templates is behind a feature flag. 

## Testing Instructions

- First make sure that saved templates doesn't show up without flipping the feature flag.
- Enable `Enable in-progress views to be accessed` feature flag and see that the saved templates page will load. 
- You can check out the empty state, OR you can also go flip `Enable in-progress story actions` to true and then go click "Create template" in a story's context menu. There's no feedback so just navigate to saved templates and see that you now have a copy of that story as a template. 
- The only action that's enabled right now is to preview the story you just made into a template. Everything else that's a button or a list item for that story should be disabled. 
- There's some console warnings because the stories returned in the API aren't returning the proper data so the console's going to yell and I flipped a required part of the card preview actions to false to prevent bummer errors in trying to make a story out of a data-less "template".
- Also see now storybook has saved templates as a dashboards/view section 

<img width="462" alt="Screen Shot 2020-10-08 at 4 39 33 PM" src="https://user-images.githubusercontent.com/10720454/95523936-dd6a5b00-0984-11eb-990e-6c5018bc4e44.png">


---



Fixes #
